packages/dante3/src/blocks/extension.ts

```diff
@@ -12,33 +12,33 @@ export function extensionFactory(options) {
     atom: options.atom || false,
     addOptions: options.options || {},
     // priority: options.priority || 1, // somehow this option breaks the addKeyboardShortcut
-    onBeforeCreate({ editor }) {
+    onBeforeCreate() {
       // Before the view is created.
-      options.onBeforeCreate && options.onBeforeCreate(editor);
+      options.onBeforeCreate && options.onBeforeCreate(this.editor);
```

Comment:
why is needed `this.editor`? is this working?

Reply:
There are actually two ways to do this because it looks like tiptap both passes the object in and binds the function to it.

So this:

```
methodName({ editor }) {
    doThingWith(editor)
}
```

Can become:

```
methodName() {
    doThingWith(this.editor)
}
```

Or this:

```
methodName(this: { editor }) {
    doThingWith(editor)
}
```

But if you do the second one, it seems like it no longer recognizes the type of the editor argument for some reason.

Given that these functions are being bound to the object within tiptap and their this argument is very well typed, I assume this works. It also matches their documentation. I did some minimal testing to make sure I didn't massively break anything, but I'm sure it needs more attention.